lib/util/stream_utils.js

```diff
@@ -994,7 +994,13 @@ shaka.util.StreamUtils = class {
 
     // The stream width and height represent the size of the entire thumbnail
     // sheet, so divide by the layout.
-    const reference = stream.segmentIndex.get(0);
+    let reference = null;
+    // Note: segmentIndex is built by default for HLS, but not for DASH, but
+    // in DASH this information comes at the stream level and not at the
+    // segment level.
+    if (stream.segmentIndex) {
+      reference = stream.segmentIndex.get(0);
+    }
     let layout = stream.tilesLayout;
```

Comment:
It just occurred to me, reading this, that this value should probably be assigned to the `tilesLayout` of the track, below.

Reply:
Something like this?

```
let layout = stream.tilesLayout;
if (!layout) {
    // Note: segmentIndex is built by default for HLS, but not for DASH, but
    // in DASH this information comes at the stream level and not at the
    // segment level.
    if (stream.segmentIndex) {
      const reference = stream.segmentIndex.get(0);
      if (reference) {
        layout = reference.getTilesLayout();
     }
    }
}
```

Reply:
Done!